Marlin/ultralcd.cpp

```diff
@@ -2408,9 +2432,9 @@ void kill_screen(const char* lcd_msg) {
     void _reset_e_acceleration_rate(const uint8_t e) { if (e == active_extruder) _reset_acceleration_rates(); }
     void _reset_e0_acceleration_rate() { _reset_e_acceleration_rate(0); }
     void _reset_e1_acceleration_rate() { _reset_e_acceleration_rate(1); }
-    #if E_STEPPERS > 2
+    #if E_STEPPERS > 2 || (ENABLED(SINGLENOZZLE_SWITCHING_EXTRUDER) && EXTRUDERS > 2)
```

Comment:
That isn't right... In the case aof singlenozzle switching extruder, you have 1 E_STRPPER but 2 EXTRUDER or 
2 E_STEPPERS and 4 EXTRUDER. So if you want to control all 4 Extruder differently, you must use EXTRUDER not  E_STEPPERS to match al 4 posibble settings (example: differnt material, that use more stepps , like rubber)